netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -202,7 +203,11 @@ public long unread() throws IOException {
             )
             .where(DyFriends.HASH, Conditions.equalTo(this.bout))
             .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
+            .iterator();
+        if (!iterator.hasNext()) {
+            throw new Inbox.BoutNotFoundException(this.bout);
```

Comment:
@2686747 I agree 100%, but then I'd suggest to just add a puzzle for cleaning this up.
Look, it's just that we settled on not adding a test here under the assumption of a small change.
=> can't make a big one :)
=> I think my proposal, though suffering some theoretical (philosophical :P) weaknesses, is the smallest possible change, same as the initial `RsFailure` approach and has the advantage of working the exact same way another "Bout not found" spot already works.

Really we shouldn't have refactoring mixed with fixes, my suggested fix is the smallest change and safest at that I can think of, given that we're not adding a test here ... I really don't see a downside :)